packages/plugins/other/visitor-plugin-common/src/imports.ts

```diff
@@ -56,9 +57,14 @@ export function generateImportStatement(statement: ImportDeclaration): string {
     importSource.identifiers && importSource.identifiers.length
       ? `{ ${Array.from(new Set(importSource.identifiers)).join(', ')} }`
       : '*';
+  const importExtension =
+    importPath.startsWith('/') || importPath.startsWith('.') ? (statement.emitLegacyCommonJSImports ? '' : '.js') : '';
   const importAlias = importSource.namespace ? ` as ${importSource.namespace}` : '';
   const importStatement = typesImport ? 'import type' : 'import';
-  return `${importStatement} ${importNames}${importAlias} from '${importPath}';${importAlias ? '\n' : ''}`;
+  return `${importStatement} ${importNames}${importAlias} from '${importPath}${importExtension}';${
```

Comment:
@saihaj I saw that we are adding `.js` to `.d.ts` file imports, would it actually work?

ex: `import * as Types from '../types.d.js';` seems invalid to me

In short, do we need to add `.js` to `.d.ts` imports?

_/invoke @n1ru4l 👀_

Reply:
```
import * as Types from '../types.d.js';
```

is legit according to TypeScript module resolution `d.js` will point to `d.ts`. It is super confusing same as `.js` points to `.ts`, but that is how they decided upon.

Reply:
Okay, thanks @n1ru4l!